torch/testing/_internal/common_modules.py

```diff
@@ -3028,14 +3028,20 @@ def module_inputs_torch_nn_ConstantPad3d(module_info, device, dtype, requires_gr
                module_inputs_func=module_inputs_torch_nn_ReflectionPad2d,
```

Comment:
Can you double check that these test inputs covers good channels last specific test cases?

I'm worried that they might work well for generic strided inputs, but didn't take into account also exercising the channels last format.

Reply:
right now, all the memory format related tests has been re-structured into `test_memory_format` from `test_modules.py`, it will check:
* output memory format (given nhwc input will get nhwc output)
* mixed memory format (some OPs can have different memory format for input and weight)
* correctness, compare the output of contiguous and channels last

So if I intentionally let the channels last kernel generate `wrong` output, it will fail the following test cases such as below:
```
(pytorch-mingfei) [mingfeim@mlt-skx091 test]$ python test_modules.py TestModuleCPU.test_memory_format_nn_ReflectionPad2d_cpu_float32
F
======================================================================
FAIL: test_memory_format_nn_ReflectionPad2d_cpu_float32 (__main__.TestModuleCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 414, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_modules.py", line 118, in test_wrapper
    return test(*args, **kwargs)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_cuda.py", line 162, in wrapped
    return f(*args, **kwargs)
  File "test_modules.py", line 695, in test_memory_format
    self.assertEqual(outputs, desired_outputs, rtol=rtol, atol=atol)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_utils.py", line 3100, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Tensor-likes are not close!

Mismatched elements: 1008 / 1296 (77.8%)
Greatest absolute difference: 16.82485580444336 at index (0, 0, 1, 1) (up to 1e-05 allowed)
Greatest relative difference: 143.67701721191406 at index (0, 1, 0, 3) (up to 1e-05 allowed)

----------------------------------------------------------------------
Ran 1 test in 0.013s

FAILED (failures=1)
(pytorch-mingfei) [mingfeim@mlt-skx091 test]$ python test_modules.py TestModuleCPU.test_memory_format_nn_ReflectionPad3d_cpu_float32
F
======================================================================
FAIL: test_memory_format_nn_ReflectionPad3d_cpu_float32 (__main__.TestModuleCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_device_type.py", line 414, in instantiated_test
    result = test(self, **param_kwargs)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_modules.py", line 118, in test_wrapper
    return test(*args, **kwargs)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_cuda.py", line 162, in wrapped
    return f(*args, **kwargs)
  File "test_modules.py", line 695, in test_memory_format
    self.assertEqual(outputs, desired_outputs, rtol=rtol, atol=atol)
  File "/home/mingfeim/pytorch/pytorch/torch/testing/_internal/common_utils.py", line 3100, in assertEqual
    raise error_metas[0].to_error(
AssertionError: Tensor-likes are not close!

Mismatched elements: 1650 / 1944 (84.9%)
Greatest absolute difference: 17.179780960083008 at index (0, 0, 5, 1, 3) (up to 1e-05 allowed)
Greatest relative difference: 3617.848876953125 at index (1, 1, 3, 5, 0) (up to 1e-05 allowed)

----------------------------------------------------------------------
Ran 1 test in 0.012s

FAILED (failures=1)

```

Reply:
@cpuhrsch could you please help review this one , thanks !